src/System.Memory/tests/ReadOnlySpan/Overlaps.cs

```diff
@@ -0,0 +1,972 @@
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class ReadOnlySpanTests
```

Comment:
Do you think it would be an interesting case to add tests as outerloop for really large spans (close to integer overflow)?

Reply:
Yes. At least testing once with spans with int.MaxValue elements would be a good idea.